cosmo/monitor_helpers.py

```diff
@@ -29,22 +37,92 @@ def explode_df(df: pd.DataFrame, list_keywords: Union[list, tuple]) -> pd.DataFr
 
 
-def compute_absolute_time(df: pd.DataFrame) -> Tuple[Time, Time]:
-    """Given a dataframe with EXPSTART keyword and a TIME column, compute the absolute time for the TIME column defined
-    as EXPSTART + TIME[i] for each element, i in TIME.
+class AbsoluteTime:
+    """Class that encapsulates the computation of an 'absolute time' for COS Data, where the 'absolute time' is defined
+    as the data's time array (assumed to be relative to the start of the exposure, t=0s) relative to the file's
+    EXPSTART [mjd].
     """
-    if 'EXPSTART' not in df or 'TIME' not in df:
-        raise KeyError('To compute the absolute time, EXPSTART and TIME must be present in the dataframe.')
+    def __init__(self, df: pd.DataFrame = None, expstart: ARRAY[float] = None, time_array: ARRAY = None,
+                 time_array_key: str = None):
+        """Initalize AbsoluteTime from a dataframe or arrays.
+        Optionally provide a time_array_key keyword if ingesting from a dataframe which contains a time array with a
+        different name from 'TIME', or if the dataframe contains multiple 'time' columns.
+        """
+        self.df = df
+        self.expstart = expstart
+        self.time = time_array
+        self.time_key = time_array_key
+        self.expstart_time = None
+
+        if self.df is not None:
+            if self.expstart is not None or self.time is not None:
+                raise ValueError('Can use a dataframe or the expstart and time_array arguments, not both.')
+
+            self._ingest_df()
+
+        if (self.expstart is not None and self.time is None) or (self.expstart is None and self.time is not None):
+            raise ValueError('If expstart and time_array must be used together.')
+
+        if self.df is None and self.expstart is None and self.time is None:
+            raise ValueError('Computing and absolute time requires either a dataframe or set of arrays')
+
+    def _ingest_df(self):
+        if 'EXPSTART' not in self.df:
+            raise KeyError('Could not find an EXPSTART column.')
+
+        self.expstart = self.df.EXPSTART
+
+        if not self.time_key:
+            try:
+                self.time = self.df.TIME
+
+            except (KeyError, AttributeError):
+                raise KeyError(
+                    'Could not find a TIME column. Try setting the time_key attribute if the TIME column has a '
+                    'different name.'
+                )
+
+        else:
+            self.time = self.df[self.time_key]
+
+    def compute_absolute_time(self, time_delta_format: str = 'sec') -> TimeDelta:
+        self.expstart_time = Time(self.expstart, format='mjd')
+        time_delta = TimeDelta(self.time, format=time_delta_format)
+
+        return self.expstart_time + time_delta
+
+    @classmethod
+    def from_df(cls, df: pd.DataFrame, time_array_key: str = None):
+        """Initialize from a dataframe."""
+        return cls(df=df, time_array_key=time_array_key)
```

Comment:
Have you only made these classmethods so that you can instantiate from ``cls`` or because you care about... inheritance or something?

I ask because it looks like they're just pseudonyms and should be staticmethods instead.

Though, TBH I'm not sure why the exist at all?

Reply:
It's a mix of alternative constructors and performing the computation without creating a new instance of the class at all. 

I wasn't sure that this was the best idea, and this may be better as a function rather than as a class

Reply:
``cls()`` is calling the existing class's constructor which means that it can already handle these diff data types, so why would the user not just instantiate using the actual class name? So there are two questions:
 * why do this at all
 * why not a staticmethod, was inheritance a concern?

Reply:
>I wasn't sure that this was the best idea, and this may be better as a function rather than as a class

My suggestion is still a func (or not at all), I don't get when any scenario would be another class?